packages/core/src/error-reporter.ts

```diff
@@ -166,4 +174,30 @@ export class ErrorReporter {
 		if (typeof e === 'string') return new ApplicationError(e);
 		return;
 	}
+
+	/** @returns true if the error should be filtered out */
+	private handleSqliteError(error: unknown) {
+		if (
+			error instanceof Error &&
+			error.name === 'QueryFailedError' &&
+			['SQLITE_FULL', 'SQLITE_IOERR'].some((errMsg) => error.message.includes(errMsg))
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/** @returns true if the error should be filtered out */
```

Comment:
I find it more intuitive to think of `isIgnored*Error` rather than thinking of `true` leading to exclusion in a `handle*` named method.